src/NServiceBus.Core.Tests/Recoverability/When_message_is_deferred_by_slr.cs

```diff
@@ -0,0 +1,145 @@
+﻿namespace NServiceBus.Core.Tests.Recoverability
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Transports;
+    using NUnit.Framework;
+    using Testing;
+
+    [TestFixture]
+    public class When_message_is_deferred_by_slr
```

Comment:
Since this is our first go at these kinds of tests I'm good with merging it as is, except maybe the underscores at the beginning of the test name ;)

Although I think we could leverage a bit more the nature of this test and test more stuff that is set for the delayed delivery part (headers, context properties...)